tikv-client/src/main/java/com/pingcap/tikv/types/IntegerType.java

```diff
@@ -61,6 +63,25 @@ private static BigDecimal unsignedValueOf(long x) {
     return new BigDecimal(UnsignedLong.fromLongBits(x).bigIntegerValue());
   }
 
+  @Override
+  protected Object doConvertToTiDBType(Object value)
+      throws ConvertNotSupportException, ConvertOverflowException {
+    // TODO: support write to YEAR
+    if (this.getType() == MySQLType.TypeYear) {
+      throw new ConvertNotSupportException(value.getClass().getName(), this.getClass().getName());
+    }
+
+    Long result;
```

Comment:
what if we want to convert a non-number object to long? e.g "1000", How do we handle this?

Reply:
see UT `Test Convert from String to SIGNED`